MediaBrowser.Providers/MediaInfo/FFProbeAudioInfo.cs

```diff
@@ -102,71 +104,79 @@ protected void Fetch(Audio audio, Model.MediaInfo.MediaInfo mediaInfo, Cancellat
+            if (tags != null){
+                if (audio.SupportsPeople && !audio.LockedFields.Contains(MetadataField.Cast))
                 {
-                    PeopleHelper.AddPerson(people, new PersonInfo
+                    var people = new List<PersonInfo>();
+                    foreach (var albumArtist in tags.AlbumArtists)
                     {
-                        Name = person.Name,
-                        Type = person.Type,
-                        Role = person.Role
-                    });
-                }
-
-                _libraryManager.UpdatePeople(audio, people);
-            }
-
-            audio.Album = data.Album;
-            audio.Artists = data.Artists;
-            audio.AlbumArtists = data.AlbumArtists;
-            audio.IndexNumber = data.IndexNumber;
-            audio.ParentIndexNumber = data.ParentIndexNumber;
-            audio.ProductionYear = data.ProductionYear;
-            audio.PremiereDate = data.PremiereDate;
+                        PeopleHelper.AddPerson(people, new PersonInfo
+                        {
+                            Name = albumArtist,
+                            Type = "AlbumArtist"
+                        });
+                    }
 
-            // If we don't have a ProductionYear try and get it from PremiereDate
-            if (audio.PremiereDate.HasValue && !audio.ProductionYear.HasValue)
-            {
-                audio.ProductionYear = audio.PremiereDate.Value.ToLocalTime().Year;
-            }
+                    foreach (var performer in tags.Performers)
+                    {
+                        PeopleHelper.AddPerson(people, new PersonInfo
+                        {
+                            Name = performer,
+                            Type = "Artist"
+                        });
+                    }
+
+                    _libraryManager.UpdatePeople(audio, people);
+                    audio.Artists = tags.Performers;
+                    audio.AlbumArtists = tags.AlbumArtists;
+                }
 
-            if (!audio.LockedFields.Contains(MetadataField.Genres))
-            {
-                audio.Genres = Array.Empty<string>();
+                audio.Name = tags.Title;
+                audio.Album = tags.Album;
+                audio.IndexNumber = Convert.ToInt32(tags.Track);
+                audio.ParentIndexNumber = Convert.ToInt32(tags.Disc);
+                audio.ProductionYear = Convert.ToInt32(tags.Year);
+                audio.PremiereDate = Convert.ToDateTime("01/01/" + tags.Year, CultureInfo.InvariantCulture);
 
-                foreach (var genre in data.Genres)
+                if (!audio.LockedFields.Contains(MetadataField.Genres))
                 {
-                    audio.AddGenre(genre);
+                    audio.Genres = Array.Empty<string>();
+
+                    foreach (var genre in tags.Genres)
+                    {
+                        audio.AddGenre(genre);
+                    }
```

Comment:
```suggestion
                    audio.Genres = tags.Genres.Distinct(StringComparer.OrdinalIgnoreCase).ToArray();
```

This should be more performant and allocate less?